client/allocrunner/groupservice_hook_test.go

```diff
@@ -16,6 +18,13 @@ var _ interfaces.RunnerPrerunHook = (*groupServiceHook)(nil)
 var _ interfaces.RunnerUpdateHook = (*groupServiceHook)(nil)
 var _ interfaces.RunnerPostrunHook = (*groupServiceHook)(nil)
 
+type testRestartCounter int
```

Comment:
TODO(Nick): either remove and use `agentconsul.NoopRestarter()` or export `agentconsul.restartTracker`